### ISSUE TITLE
[Bug]: Lizmap 3.8 regression : tooltip HTML is not properly rendered in popups anymore

### ISSUE THREAD

**Issue by vlebert** [3liz/lizmap-web-client#4990]
### What is the bug? (in English)

Since 3.8 updates, HTML in QGIS tooltip does not seem to be rendered properly.

For exemple this tooltip used to be rendered properly. In 3.8

- The HTML is broken (why `<ul>` in not inside `<p>` anymore)
- The `<script>` part is missing

![digi-studio 2024-11-18 at 16 48 04](https://github.com/user-attachments/assets/fe2ec8e0-d157-4cf1-a012-e5d22fe1f783)

### Steps to reproduce the issue

Just load the attached project in Lizmap 3.8

[lizmap_regression_popup.zip](https://github.com/user-attachments/files/17803233/lizmap_regression_popup.zip)

It contains a layer with popup HTML configured as

```html
<p>Hello</p>
<p>
<ul>
<li>liste 1</li>
<li>liste 2</li>
</ul>
</p>
<script>console.log("test script")</script>
```

it renders as

```html
<p>Hello</p>


<p>
</p>
<ul>
<li>liste 1</li>
<li>liste 2</li>
</ul>
<p></p>
```

And the script is removed.
 

### Versions, safeguards, check summary etc

**Versions :**

* Lizmap Web Client : 3.8.4-pre.8065
* Lizmap plugin : 4.4.4
* QGIS Desktop : 3.34.6
* Lizmap.com : Yes
* QGIS Server : 3.34.11
* Py-QGIS-Server : 1.9.2
* QGIS Server plugin DataPlotly : Not installed
* QGIS Server plugin atlasprint : 3.4.1
* QGIS Server plugin cadastre : 1.20.0
* QGIS Server plugin lizmap_server : 2.11.2
* QGIS Server plugin wfsOutputExtension : 1.8.2

<details>
<summary>List of Lizmap Web Client modules :</summary>
<br/>
* saas : 1.7.4
* webdav : 1.1.6
</details>

<details>
<summary>List of safeguards :</summary>
<br/>
* Mode : normal<br/>
* Allow parent folder : no<br/>
* Prevent other drive : yes<br/>
* Prevent PG service : no<br/>
* Prevent PG Auth DB : yes<br/>
* Force PG user&pass : yes<br/>
* Prevent ECW : yes<br/>
</details>

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

### Operating system

Lizmap hosting

### Browsers

Chrome

### Browsers version

Version 1.69.0 (55816)

**Comment by Gustry** [comment]
> The <script> part is missing

This is expected now. See for example #4914 or #4707 It wasn't advised, even before 3.8 to use `<script>` inside a popup.

This was a security issue, related to [XSS](https://fr.wikipedia.org/wiki/Cross-site_scripting)

> The HTML is broken (why <ul> in not inside <p> anymore)

Are you sure it's the same `<p>` ? Very quick debug, can you try to add `<p class="test">` jsut to check in the meantime ?

**Comment by Gustry** [comment]
Indeed, there is a difference in the HTML, for instance in QGIS HTML maptip : 

```html
<p>First P block 1</p>
<p>
Begin new P block 2
<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
End P block 2
</p>
```

It gives in LWC :

```html
<p>First P block 1</p>
<p>
Begin new P block 2
</p><ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
End P block 2
<p></p>
```

**Comment by vlebert** [comment]
Thanks for the information. 
Well I see that javascript is not supported anymore inside popups, but IO don't see what does it has to do with XSS we are not doing any cross site calls. 

The `<script>` is directly rendered in the HTML document and not embedded as an iframe or something. There should not be any issues with XSS am I wrong ?

By the way, do iframe still work ?

Anyway, how am I supposed to reproduce this behaviour with Lizmap Javascript signals (https://docs.lizmap.com/current/en/publish/customization/javascript.html#available-javascript-events)

For example, `lizmappopupdisplayed` does not return the feature ID so how to have a custom function for specific popups ? Do you have any example ?

What I am trying to do is to render a specific custom code inside a popup. Previously I used to work with iframes but this time I tried to do things cleaner with native Lizmap JS/HTML. 

Should I continue to embed external content ?

**Comment by Gustry** [comment]
The security issue was to load external JavaScript which was not part of the core main Lizmap Web Client application and the dedicated `media/js` folder. (Javascript from a GIS file as well for instance)

> For example, `lizmappopupdisplayed` does not return the feature ID so how to have a custom function for specific popups ? Do you have any example ?

I was curious as well and I look on GitHub : either in this repository, or in the javascript-script-library repository : 

* https://github.com/search?q=repo%3A3liz%2Flizmap-web-client+lizmappopupdisplayed&type=code
* https://github.com/search?q=repo%3A3liz%2Flizmap-javascript-scripts%20lizmappopupdisplayed&type=code

I tried this just now, I will let other comment if there is something : 

```javascript
lizMap.events.on({
    lizmappopupdisplayed: function (popup) {
        let popupContainer = document.getElementById(popup.containerId);
        if (!popupContainer) return false;

        Array.from(popupContainer.querySelectorAll('div.lizmapPopupContent input.lizmap-popup-layer-feature-id')).map(element => {
            let val = element.value;
            let featureId = val.split('.').pop();
            let layerId = val.replace('.' + featureId, '');
            console.log(featureId)
            console.log(layerId)
        });
    }
});
```

> Should I continue to embed external content ?

I'm not sure what you mean by "external content", but iframe is supported.

**Comment by Gustry** [comment]
I made a PR to make something a little bit easier #4992

**Comment by vlebert** [comment]
Thanks

**Comment by vlebert** [comment]
I could fix my popup thanks to your help.

I am wondering what do you mean exactly in 3.8 changelog by 

> Nous vous recommandons fortement d'utiliser la nouvelle API JavaScript lors de l'écriture d'un script JavaScript complémentaire personnalisé.

I visited this doc, but it does not help me : https://docs.3liz.org/lizmap-web-client/js/module-FeatureToolbar.html
